cmd/api/controller/controller.go

```diff
@@ -17,18 +42,39 @@ func NewPessoaController(service pessoa.Service) *PessoaController {
 }
 
 func (c *PessoaController) Create(ctx *fiber.Ctx) error {
-	var pessoa *domain.Pessoa
+	pessoareq := PessoaRequest{}
 
-	if err := ctx.BodyParser(&pessoa); err != nil {
-		return err
+	validate := validator.New()
+
+	if err := ctx.BodyParser(&pessoareq); err != nil {
+		return fiber.NewError(http.StatusBadRequest, fmt.Sprintf("invalid json: %s", err.Error()))
+	}
+
+	if err := validate.Struct(&pessoareq); err != nil {
+		return fiber.NewError(http.StatusBadRequest, err.Error())
 	}
 
-	err := c.service.CreatePessoa(pessoa)
+	newPessoa := domain.Pessoa{
+		CpfCnpj:    pessoareq.CpfCnpj,
+		Name:       pessoareq.Name,
+		Nascimento: pessoareq.Nascimento,
+		Seguros:    pessoareq.Seguros,
+	}
+
+	pessoaSucess, err := c.service.CreatePessoa(&newPessoa)
```

Comment:
Isso é frescuragem minha, então sinta-se avontade para deixar como está.

Mas geralmente eu evito dar nomes a coisas que só serão usadas uma única vez. Eu acho que isso facilita a leitura do código um pouco, aí eu faria isso assim:

```suggestion
	pessoaSucess, err := c.service.CreatePessoa(&domain.Pessoa{
		CpfCnpj:    pessoareq.CpfCnpj,
		Name:       pessoareq.Name,
		Nascimento: pessoareq.Nascimento,
		Seguros:    pessoareq.Seguros,
	})
```